Analytics/Request/BlockingRequestHandler.cs

```diff
@@ -225,19 +223,32 @@ public async Task MakeRequest(Batch batch)
                     if (_client.Config.CompressRequest)
                         content.Headers.ContentEncoding.Add("gzip");
 
-                    var response = await _httpClient.PostAsync(uri, content).ConfigureAwait(false);
+                    HttpResponseMessage response = null;
+                    bool retry = false;
+                    try
+                    {
+                        response = await _httpClient.PostAsync(uri, content).ConfigureAwait(false);
+                    }
+                    catch (System.Threading.Tasks.TaskCanceledException)
```

Comment:
We may want to consider manually handling cancellation on timeout with `PostAsync`  so we can throw the right kind of exception (if we control the timeout, we can throw a `TimeoutException` instead), but that might be out of scope for this change.

Something like: 
https://www.thomaslevesque.com/2018/02/25/better-timeout-handling-with-httpclient/

Thoughts @f2prateek @nhi-nguyen ?

Reply:
Nice find! Yeah I think it's a bit out of scope for this change, but definitely let's plan to tackle this. I'll create a JIRA ticket

Reply:
ticket - https://segment.atlassian.net/browse/LIB-670